client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogmaRepository.java

```
@@ -298,6 +352,72 @@ public WatcherRequest<Revision> watcher(PathPattern pathPattern) {
+        }
+        return commit("Import " + dir.getFileName(), changes)
+                .push(Revision.HEAD)
+                .thenApply(ImportResult::fromPushResult);
+    }
+
+    public CompletableFuture<ImportResult> importResourceDir(String dir) {
+        requireNonNull(dir, "dir");
+        return importResourceDir(dir, CentralDogmaRepository.class.getClassLoader());
+    }
+
+    public CompletableFuture<ImportResult> importResourceDir(String dir, ClassLoader classLoader) {
+        requireNonNull(dir, "dir");
+        requireNonNull(classLoader, "classLoader");
+
+
+        final URL url = requireNonNull(classLoader.getResource(dir),
+                                       () -> "resource not found: " + dir);
+        if (!"file".equals(url.getProtocol())) {
+            return CompletableFutures.exceptionallyCompletedFuture(
+                    new IllegalArgumentException("Resource dir must be explodable (got " + url + ')'));
+        }
+
+        final Path logical = Paths.get(dir);
+        final Path physicalPath = Paths.get(url.getPath());
+        if (logical.getNameCount() < 2) {
+            return CompletableFutures.exceptionallyCompletedFuture(
+                    new IllegalArgumentException("Path must be <project>/<repo>[/…]: " + dir));
+        }
+        centralDogma.createProject(projectName()).join();
+        centralDogma.createRepository(projectName(), repositoryName()).join();
+
+        final List<Change<?>> changes = new ArrayList<>();
+        if (Files.isRegularFile(physicalPath)) {
+            final String repoPath = '/' + logical.getFileName()
+                                                 .toString()
+                                                 .replace(File.separatorChar, '/');
+            changes.add(toChange(repoPath, physicalPath));
+        } else {
+            changes.addAll(collectImportFiles(physicalPath));
+        }
+
+        if (changes.isEmpty()) {
```

I think we should raise an exception in this case

---

Yeah, let's do that. 👍